deeplinkdispatch-base/src/main/java/com/airbnb/deeplinkdispatch/SchemeHostAndPath.kt

```diff
@@ -0,0 +1,23 @@
+package com.airbnb.deeplinkdispatch
+
+import com.airbnb.deeplinkdispatch.base.MatchIndex
+
+/**
+ * Used to encapsulate the scheme host and path of a DeepLinkUri into a type.
+ *
+ * All operations are happening on the UI thread by definition so it is ok to make the lazies
+ * not thread safe for speed improvement.
+ */
+class SchemeHostAndPath(val uri: DeepLinkUri) {
+    val schemeHostAndPath by lazy(LazyThreadSafetyMode.NONE) {
+        uri.scheme() + "://" + uri.encodedHost() + uri.encodedPath()
+    }
+
+    val matchList by lazy(LazyThreadSafetyMode.NONE) {
```

Comment:
In the current code, this seems like it is always used whenever a `SchemeHostAndPath`, so might not need to be lazy.